src/routes/run.rs

```diff
@@ -537,6 +492,50 @@ async fn delete_by_id(req: HttpRequest, pool: web::Data<db::DbPool>) -> impl Res
     })
 }
 
+/// Generates csv files from `runs` and returns a zip of the csvs as bytes.  If an error occurs,
+/// logs the error and returns an appropriate error response
+fn get_bytes_for_csv_zip_from_runs(
+    runs: &Vec<RunWithResultsAndErrorsData>,
+) -> Result<Vec<u8>, HttpResponse> {
+    // Build csv files from results
+    let run_csvs_dir: TempDir = match run_csv::write_run_data_to_csvs_and_zip_in_temp_dir(&runs) {
```

Comment:
The files are fairly small so I don't anticipate them taking up a bunch of disk space.  At the latest, as soon as that function ends, `run_csvs_dir` goes out of scope and the directory is cleaned up.

Reply:
That said, I might look into doing all of the csv and zip creation in memory because it seems like that might be a more efficient approach.  Thoughts?

Reply:
The idea of a bunch of threads all writing a bunch of potentially large csv files (what if i constructed a query that hit everything in the database in csv format?) to disk somewhere and them potentially clobbering eachother sounds scary. In memory certianly sounds safer though i guess at its core it doesn't fix anything.